src/contexts/AuthProvider.js

```diff
@@ -18,8 +14,8 @@ export function AuthProvider({ children }) {
 
   async function signup(email, password, uname) {
     return auth.createUserWithEmailAndPassword(email, password).then((userData) => {
-      emailLogin(email, password).then(() => {
-        db.collection('users').doc(userData.user.uid).set({
+      emailLogin(email, password).then(async () => {
+        await db.collection('users').doc(userData.user.uid).set({
```

Comment:
for firestore collections, let's add the collection names in a constants file. `constants/collections`